deps/src/polymake_integers.cpp

```diff
@@ -6,19 +6,31 @@
 
 #include "polymake_integers.h"
 
-pm::Integer new_integer_from_bigint(jl_value_t* integer){
+pm::Integer cast_bigint_to_integer(jl_value_t* integer){
```

Comment:
Please note that this is not a cast, but indeed copies the integer. Everything else is unsafe.

The copy is implicit in the way CxxWrap works: by returning the dereferenced `p` CxxWrap actually needs to create a copy of `p`, and therefore the integer.

Reply:
Oh good to know. What is about the definition on the Julia side? Does this copy?

Reply:
```julia
"""
    bigint!(int::Polymake.pm_IntegerAllocated)

Cast a `Polymake.pm_IntegerAllocated` to a `BigInt`. This *does not* create a copy.
"""
function bigint!(int::Polymake.pm_IntegerAllocated)
    unsafe_load(reinterpret(Ptr{BigInt},int.cpp_object))
end
```

Reply:
No, it does not.

`unsafe_load` in Julia just gives you an object ref to whatever the pointer points to. It would be safer to actually create a copy there.